app/javascript/controllers/expandcollapse_controller.js

```diff
@@ -0,0 +1,17 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  toggleDescriptions(evt) {
+    const activeClass = 'active'
+    const currentActive = this.element.querySelector(`.${activeClass}`)
+    currentActive.classList.remove(activeClass)
+    currentActive.disabled = false;
+    const target = evt.target.classList.contains('bi') ? evt.target.parentElement : evt.target;
+    target.disabled = true;
+    target.classList.add(activeClass)
+    const elements = Array.from(document.getElementsByClassName('more-info-area'));
```

Comment:
This is a pretty fragile way to bind to these elements.  Have you considered making them stimulus targets for this controller?  Alternatively, you could use stimulus outlets to reference another controller responsible for each document. https://stimulus.hotwired.dev/reference/outlets or even just emit an custom event like "showList"/"showParagraph", that another controller will bind to.

Reply:
I thought about the stimulus targets but then you have to figure out which got clicked. I will think about it some more maybe.

Reply:
I still think that binding behaviors to css styles (`.more-info-area`) is not very robust.  I urge you to look at the solutions that are built into Stimulus for this.

Reply:
Gotcha I misunderstood the area the comment was pointing towards, for some reason my brain said the top line and not the last line. Makes sense.